src/load_tests/treehub.cc

```diff
@@ -0,0 +1,103 @@
+#include "treehub.h"
+#include <gio/gio.h>
+#include <glib.h>
+#include <ostree.h>
+#include <package_manager/ostreemanager.h>
+#include <random>
+#include <vector>
+#include "context.h"
+#include "executor.h"
+
+namespace fs = boost::filesystem;
+
+class FetchTask {
+  const Config &config;
+  std::shared_ptr<INvStorage> storage;
+  std::shared_ptr<KeyManager> keys;
+  const fs::path repoDir;
+  const std::string branchName;
+  const std::string remoteUrl;
+
+  OstreeRepo *repo = nullptr;
+  void initRepo() {
+    GFile *gRepoFile = g_file_new_for_path(repoDir.native().c_str());
```

Comment:
I know these are "just" tests, but still, for better memory management, can we use `StructGuard` here? There are some examples in `src/libaktualizr/package_manager/ostreemanager.cc`.